apps/spark/src/main/java/com/linkedin/openhouse/jobs/client/TablesClient.java

```diff
@@ -89,6 +89,20 @@ protected GetTableResponseBody getTable(TableMetadata tableMetadata) {
         null);
   }
 
+  /**
+   * Checks if data layout strategies can be generated on the input table.
+   *
+   * @param tableMetadata table metadata
+   * @return true if data layout strategies can be generated on the table, false otherwise
+   */
+  public boolean canRunDataLayoutStrategyGeneration(TableMetadata tableMetadata) {
+    GetTableResponseBody response = getTable(tableMetadata);
+    if (response == null || response.getTimePartitioning() == null) {
```

Comment:
partitioning table is excluded ?

Reply:
It's the opposite, although if the response is empty, we skip too, so sometimes we skip partitioned tables

Reply:
Need to handle this as well ?https://github.com/linkedin/openhouse/blob/83b3797c7c278345457d9b43ac16ce1e26a9ceab/services/tables/src/main/java/com/linkedin/openhouse/tables/api/spec/v0/response/GetTableResponseBody.java#L91

Reply:
good catch, fixed